### DISCUSSION TITLE
Values of Smart Meter Interface to KNX

### DISCUSSION THREAD

**Discussion by empty88** [arendst/Tasmota#15585] [Q&A] [answered]
Hello everybody,
is there a way to send values received by a SML Script to KNX?

I am using the following script and sending values over mqtt works. Now i would like to send these values to KNX.

> \>D
\>B
->sensor53 r
tper=10
\>W
\>M 1
+1,4,s,1,9600,Strom
;1,77078181C78203FF@#,Zähler,,Meter_id,0
1,77070100010800FF@1000,Zählerstand,kWh,Power,2
1,77070100100700FF@1,Aktuell,W,Power_curr,0
1,77070100020800FF@1000,Einspeisung,kWh,,2
\#

best regards

**Comment by gemu2015** [comment, marked as answer]
i don't use knx but i added a knx(code value) function to scripter which you may try.
you should check the scripting and SML docs how to get SML values into scripter (e.g. sml[x])
then check in Tasmota.h the Sensor and Commands definition for KNX Driver e.g. KNX_ENERGY_POWER being 21

then on every teleperiod transfer current power

\>S
if upsecs%tper==0 {
knx(21 sml[2])
}

[xdrv_10_scripter.ino.zip](https://github.com/arendst/Tasmota/files/8680442/xdrv_10_scripter.ino.zip)

**Comment by elelom2** [comment]
Leider bin ich zu blöd, das oben genante sinnvoll umzusetzen.
Wunsch: 2 bis 5 Zählerwerte aus dem Smartmeter an ein KNX-Gerät schreiben.
SML-Script für den Zähler funktioniert, sehe alle Daten in der Tasmota-Seite.
Button/Toggle auf den KNX-Bus schreibt auch, Kommunikation zum Bus ist also auch i.O.
Aber wie muss die Syntax sein um einen Zählerwert aus den Script Richtung Bus zu senden?
Wo finde ich ggf ein Beispielwie eine Variable aus dem Script gesendet wird.
Danke Jens

die modifizierte xdrv_10...ino habe ich eingefügt.
